app/api/logging/audit.py

```diff
@@ -20,54 +20,117 @@
+    # Define this variable locally so it can't be modified by other modules.
+    #
+    # Events from the suggested audit hook locations in PEP 578 that we aren't logging:
+    #
+    # compile and import
+    #     Detects when code is being compiled or imported.
+    #
+    #     Why we aren't logging:
+    #     Logging as part of regular imports is too noisy.
+    #
+    # sys._getframe
+    #     Detects when code is accessing frames directly.
+    #
+    #     Why we aren't logging:
+    #     The logging module itself calls getFrame
+    #     (see https://github.com/python/cpython/blob/bd7903967cd2a19ebc842dd1cce75f60a18aef02/Lib/logging/__init__.py#L170)
+    #     so logging on this event would trigger an infinite loop unless
+    #     logging._srcfile is set to None
+    #     See https://github.com/python/cpython/blob/bd7903967cd2a19ebc842dd1cce75f60a18aef02/Lib/logging/__init__.py#L179-L189
+    #
+    # object.__getattr__
+    #     Detect access to restricted attributes. This event is raised for any built-in members
+    #     that are marked as restricted, and members that may allow bypassing imports.
+    #
+    #     Why we aren't logging:
+    #     The logging module itself accesses frame.f_code.co_filename
+    #     (see https://github.com/python/cpython/blob/bd7903967cd2a19ebc842dd1cce75f60a18aef02/Lib/logging/__init__.py#L202)
+    #     which calls object.__getattr__ under the hood
+    #     (see https://github.com/python/cpython/blob/main/Objects/frameobject.c#L89-L95)
+    #
+
+    EVENTS_TO_LOG = {
+        # Detect when new audit hooks are being added.
+        "sys.addaudithook": (),
+        # Detects any attempt to set the open_code hook.
+        "cpython.PyFile_SetOpenCodeHook": (),
+        # Detect dynamic execution of code objects. This only occurs for explicit
+        # calls, and is not raised for normal function invocation.
+        "exec": ("code_object",),
+        # Detect when a file is about to be opened. path and mode are the usual
+        # parameters to open if available, while flags is provided instead of
+        # mode in some cases.
+        "io.open": ("path", "mode", "flags"),
+        # Detect when code is injecting trace functions. Because of the
+        # implementation, exceptions raised from the hook will abort the
+        # operation, but will not be raised in Python code. Note that
+        # threading.setprofile eventually calls this function, so the event
+        # will be audited for each thread.
+        "sys.setprofile": (),
+        # Detect when code is injecting trace functions. Because of the
+        # implementation, exceptions raised from the hook will abort the
+        # operation, but will not be raised in Python code. Note that
+        # threading.settrace eventually calls this function, so the event will
+        # be audited for each thread.
+        "sys.settrace": (),
```

Comment:
Was on the exclude list as it break the PyCharm debugger.

Reply:
I'll remove since I can't think of a reason why this is important to log.

As an aside, I think we probably don't need to run audit logs locally.